packages/lambda-tiler/src/util/config.cache.ts

```diff
@@ -0,0 +1,40 @@
+import { ConfigBundled, ConfigProviderMemory } from '@basemaps/config';
+import { fsa } from '@chunkd/fs';
+import { SwappingLru } from './swapping.lru.js';
+
+class LruConfig {
+  value: ConfigProviderMemory;
+
+  constructor(mem: ConfigProviderMemory) {
+    this.value = mem;
+  }
+
+  get size(): number {
+    return JSON.stringify(this.value.toJson()).length;
+  }
+}
+
+export class ConfigCache {
+  cache: SwappingLru<LruConfig>;
+  constructor(maxSize: number) {
+    this.cache = new SwappingLru<LruConfig>(maxSize);
+  }
+
+  async getConfig(location: string): Promise<ConfigProviderMemory | null> {
+    const existing = this.cache.get(location)?.value;
+    if (existing != null) return existing;
+    try {
```

Comment:
the fetch needs to be sync, so multiple things can call this function

eg doing
```typescript
Promise.all([
 getConfig('a.json')
 getConfig('a.json')
 ])
 ```
 should not create two requests to 'a.json'